lib/Benchmark/Executor/MicrotimeExecutor.php

```diff
@@ -75,4 +82,77 @@ public function getDefaultConfig()
     {
         return array();
     }
+
+    private function resolveParameterOrder(SubjectMetadata $subject, array $parameters)
+    {
+        $isAssoc = false;
+        foreach (array_keys($parameters) as $key) {
+            if (is_string($key)) {
+                $isAssoc = true;
+                continue;
+            }
+
+            if (true === $isAssoc && is_numeric($key)) {
+                throw new \RuntimeException(sprintf(
+                    'Cannot mix numeric and string indexes in parameter set. Got: "%s"',
+                    implode('", "', array_keys($parameters))
+                ));
+            }
+        }
+
+        if ($isAssoc) {
+            $parameters = $this->mapAssocToArgs($subject, $parameters);
+        }
+
+        return $parameters;
+    }
+
+    private function mapAssocToArgs(SubjectMetadata $subject, array $parameters)
+    {
+        $args = $subject->getArguments();
+
+        if (count($args) && array_diff(array_keys($parameters), $args)) {
+            throw new \RuntimeException(sprintf(
+                'You have used an associative array for parameters. The argument names on the benchmarking ' . 
+                'subject "%s" must match the top level key names of the provided parameter set. Should be: "%s", current ' . 
+                'argument names: "%s"',
+                $subject->getName(),
+                implode('", "', array_keys($parameters)),
+                implode('", "', $args)
+            ));
+        }
+
+        if (count($args) && $diff = array_diff($args, array_keys($parameters))) {
```

Comment:
Document the fact that it is valid to have a zero function arity even if parameters are provided.